core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/MethodConfigurationComponent.scala

```diff
@@ -133,9 +143,14 @@ trait MethodConfigurationComponent {
       loadMethodConfigurationById(methodConfigurationId)
     }
 
-    def rename(workspaceContext: SlickWorkspaceContext, methodConfigurationNamespace: String, methodConfigurationName: String, newName: String): ReadWriteAction[Int] = {
-      workspaceQuery.updateLastModified(workspaceContext.workspaceId) andThen
-        findByName(workspaceContext.workspaceId, methodConfigurationNamespace, methodConfigurationName).map(_.name).update(newName)
+    // To rename a method config, we "delete" (or rather hide) the old version of the method config and create a new row for the new method config, with the
+    //  config version # incremented
+    def rename(workspaceContext: SlickWorkspaceContext, methodConfigurationNamespace: String, methodConfigurationName: String, newMethodConfig: MethodConfiguration) = {
+      uniqueResult[MethodConfigurationRecord](findByName(workspaceContext.workspaceId, methodConfigurationNamespace, methodConfigurationName)) flatMap {
+        case None => DBIO.successful(false)
```

Comment:
Yeah, so we've already looked up this method configuration [here](https://github.com/broadinstitute/rawls/pull/630/files#diff-91fe8efb10dfe91234a26e4f80039fc6R1083). If we can't find the record for it now, having just found it, then something has gone severely screwy.

Also it's a bit poo that we're looking up the method configuration from the database twice. You could change the call to `withMethodConfig` to a new `withMethodConfigRec` so you can just pass it all the way down.